### ISSUE TITLE
Remove GLES2 references in wlr_egl

### ISSUE THREAD

**Issue by emersion** [swaywm/wlroots#739]
`wlr_egl` assumes GLES2 and includes its headers.

This requires splitting `render/glapi.txt` into EGL and GLES2 extensions.

**Comment by ascent12** [comment]
>wlr_egl assumes GLES2

That's a pretty safe assumption. They're very closely related.
The only other place EGL could be used is with other versions of OpenGL.

However, when I get off my ass and try to add Vulkan support, EGL + glapi and its interactions with everything else is going to need to change a lot.

**Comment by emersion** [comment]
>The only other place EGL could be used is with other versions of OpenGL.

Yeah, that's what I'm talking about here. We may want a GLES3 renderer in the future (or somebody might want to build its own GLES3/OpenGL renderer).

>However, when I get off my ass and try to add Vulkan support, EGL + glapi and its interactions with everything else is going to need to change a lot.

Aye, we'll probably want to abstract `wlr_egl` with an interface (and rename it). This is going to be a mess.

**Comment by SirCmpwn** [comment]
Does it make sense to lump together EGL and GLES and merge the APIs

**Comment by acrisci** [comment]
> The only other place EGL could be used is with other versions of OpenGL.

[Checks out](https://www.lunarg.com/faqs/interaction-egl-vulkan/)

> The bottom line is that there is no EGL for Vulkan, rather WSI in Vulkan is handled by extensions.

**Comment by emersion** [comment]
>Does it make sense to lump together EGL and GLES and merge the APIs

I think we should keep it easy for library users to swap a renderer with their own. Also, EGL could be used with different renderers.
